spec/integration/util/windows/principal_spec.rb

```diff
@@ -103,31 +116,45 @@
   end
 
   describe ".lookup_account_sid" do
-    it "should create an instance from a well-known account SID" do
-      principal = Puppet::Util::Windows::SID::Principal.lookup_account_sid(system_bytes)
-      expect(principal.account).to eq('SYSTEM')
-      expect(principal.sid_bytes).to eq(system_bytes)
-      expect(principal.sid).to eq('S-1-5-18')
-      expect(principal.domain).to eq('NT AUTHORITY')
-      expect(principal.domain_account).to eq('NT AUTHORITY\\SYSTEM')
-
-      # Windows API LookupAccountSid behaves differently if current user is SYSTEM
-      if Puppet::Util::Windows::ADSI::User.current_user_name != 'SYSTEM'
-        account_type = :SidTypeWellKnownGroup
-      else
-        account_type = :SidTypeUser
-      end
-
-      expect(principal.account_type).to eq(account_type)
+    it "should create an instance from a user SID" do
```

Comment:
This test is net new since we didn't have any checks on `SidTypeUser`.

Unfortunately to do something that should work everywhere, this requires some "SID math" on the byte array.  Thoughts?